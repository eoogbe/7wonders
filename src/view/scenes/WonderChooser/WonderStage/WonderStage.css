/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

.WonderChooserWonderStage {
  font-size: 1rem;
  display: flex;
  width: 10em;
  height: 5em;
  margin: 0 0.25em;
  padding: 0.5em;
  border: 1px solid #000;
  border-radius: 4px;
}

.WonderChooserWonderStage-cost {
  font-size: 0.75rem;
}

.WonderChooserWonderStage-effectList {
  font-size: 1.5rem;
  align-self: center;
  margin-left: 0.333em;
}
