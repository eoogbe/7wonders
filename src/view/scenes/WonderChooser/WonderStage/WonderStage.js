/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import './WonderStage.css';

const renderCostItem = (item, idx) => <li key={idx}>{item}</li>;

const renderEffect = (effect, idx) => (
  <li key={idx}>{effect}</li>
);

const WonderStage = ({ cost, effects }) => (
  <li className="WonderChooserWonderStage">
    <ul className="WonderChooserWonderStage-cost">
      {cost.map(renderCostItem)}
    </ul>
    <ul className="WonderChooserWonderStage-effectList">
      {effects.map(renderEffect)}
    </ul>
  </li>
);

export default WonderStage;
