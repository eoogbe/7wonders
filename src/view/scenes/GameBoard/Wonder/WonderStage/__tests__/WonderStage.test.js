/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import WonderStage from '../WonderStage';

const EFFECTS = ['3 points'];

const COST = ['Stone', 'Stone'];

const PAYMENT = 0;

jest.useFakeTimers();

const mockStore = configureStore();

it('renders without crashing', () => {
  const store = mockStore({});
  const tree =
    renderer.create(
      <Provider store={store}>
        <WonderStage
          effects={EFFECTS}
          cost={COST}
          payment={PAYMENT}
          isCompleted={false}
          isNext={true}
        />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders when cannot afford payment', () => {
  const store = mockStore({});
  const tree =
    renderer.create(
      <Provider store={store}>
        <WonderStage
          effects={EFFECTS}
          cost={COST}
          payment={null}
          isCompleted={false}
          isNext={true}
        />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with coin payment', () => {
  const store = mockStore({});
  const tree =
    renderer.create(
      <Provider store={store}>
        <WonderStage
          effects={EFFECTS}
          cost={COST}
          payment={2}
          isCompleted={false}
          isNext={true}
        />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders when completed', () => {
  const store = mockStore({});
  const tree =
    renderer.create(
      <Provider store={store}>
        <WonderStage
          effects={EFFECTS}
          cost={COST}
          payment={PAYMENT}
          isCompleted={true}
          isNext={false}
        />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders when neither completed nor next', () => {
  const store = mockStore({});
  const tree =
    renderer.create(
      <Provider store={store}>
        <WonderStage
          effects={EFFECTS}
          cost={COST}
          payment={PAYMENT}
          isCompleted={false}
          isNext={false}
        />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
