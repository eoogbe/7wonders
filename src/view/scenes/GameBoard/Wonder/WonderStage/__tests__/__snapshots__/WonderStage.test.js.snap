// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when cannot afford payment 1`] = `
<li
  className="GameBoardWonderStage"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(93, 64, 55, 0.5)",
    }
  }
>
  <ul
    className="GameBoardWonderStage-cost"
  >
    <li>
      Stone
    </li>
    <li>
      Stone
    </li>
  </ul>
  <ul
    className="GameBoardWonderStage-effectList"
  >
    <li>
      3 points
    </li>
  </ul>
  <div
    className="GameBoardWonderStage-payment"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8 0-1.85.63-3.55 1.69-4.9L16.9 18.31C15.55 19.37 13.85 20 12 20zm6.31-3.1L7.1 5.69C8.45 4.63 10.15 4 12 4c4.42 0 8 3.58 8 8 0 1.85-.63 3.55-1.69 4.9z"
      />
    </svg>
  </div>
</li>
`;

exports[`renders when completed 1`] = `
<li
  className="GameBoardWonderStage"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={Object {}}
>
  <ul
    className="GameBoardWonderStage-cost"
  >
    <li>
      Stone
    </li>
    <li>
      Stone
    </li>
  </ul>
  <ul
    className="GameBoardWonderStage-effectList"
  >
    <li>
      3 points
    </li>
  </ul>
</li>
`;

exports[`renders when neither completed nor next 1`] = `
<li
  className="GameBoardWonderStage"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(93, 64, 55, 0.5)",
    }
  }
>
  <ul
    className="GameBoardWonderStage-cost"
  >
    <li>
      Stone
    </li>
    <li>
      Stone
    </li>
  </ul>
  <ul
    className="GameBoardWonderStage-effectList"
  >
    <li>
      3 points
    </li>
  </ul>
</li>
`;

exports[`renders with coin payment 1`] = `
<li
  className="GameBoardWonderStage"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(93, 64, 55, 0.5)",
    }
  }
>
  <ul
    className="GameBoardWonderStage-cost"
  >
    <li>
      Stone
    </li>
    <li>
      Stone
    </li>
  </ul>
  <ul
    className="GameBoardWonderStage-effectList"
  >
    <li>
      3 points
    </li>
  </ul>
  <div
    className="GameBoardWonderStage-payment"
  >
    -
    2
  </div>
</li>
`;

exports[`renders without crashing 1`] = `
<li
  className="GameBoardWonderStage"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(93, 64, 55, 0.5)",
    }
  }
>
  <ul
    className="GameBoardWonderStage-cost"
  >
    <li>
      Stone
    </li>
    <li>
      Stone
    </li>
  </ul>
  <ul
    className="GameBoardWonderStage-effectList"
  >
    <li>
      3 points
    </li>
  </ul>
  <div
    className="GameBoardWonderStage-payment"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
      />
    </svg>
  </div>
</li>
`;
